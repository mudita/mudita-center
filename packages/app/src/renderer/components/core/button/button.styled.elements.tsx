/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { Link, NavLink } from "react-router-dom"
import transition from "Renderer/styles/functions/transition"
import theme from "Renderer/styles/theming/theme"
import {
  backgroundColor,
  borderColor,
  borderRadius,
  fontWeight,
  textColor,
  transitionTime,
  transitionTimingFunction,
  width,
} from "Renderer/styles/theming/theme-getters"
import styled, { css } from "styled-components"
import { DisplayStyle, Size } from "./button.config"
import Icon from "Renderer/components/core/icon/icon.component"

const getSize = (size: Size) => {
  switch (size) {
    case Size.FixedSmall:
      return css`
        width: ${width("buttonSmall")};
      `
    case Size.FixedMedium:
      return css`
        width: ${width("buttonMedium")};
      `
    case Size.FixedBig:
      return css`
        width: ${width("buttonBig")};
      `
    default:
      return
  }
}

export const activeClassName = "active"

const navLinkStyles = css`
  background-color: ${backgroundColor("minor")};
  * {
    color: ${textColor("primary")};
  }

  svg {
    opacity: 1;
    transition: ${transition("opacity", undefined, "ease")};
  }
`

export const disabledPrimaryStyles = css`
  background: ${backgroundColor("disabled")};
  border: 0.1rem solid ${backgroundColor("disabled")};
`

export const disabledSecondaryStyles = css`
  border: 0.1rem solid ${borderColor("secondary")};
  color: ${textColor("secondary")};
  g {
    fill: ${textColor("accent")};
  }
`

const buttonStyles = css<{
  displaystyle: DisplayStyle
  disabled: boolean
  size: Size
}>`
  display: flex;
  align-items: center;
  justify-content: center;
  appearance: none;
  background: none;
  transition: ${transition(
      "background",
      theme.transitionTime.quick,
      theme.transitionTimingFunction.easeInOut
    )},
    ${transition(
      "color",
      theme.transitionTime.quick,
      theme.transitionTimingFunction.easeInOut
    )},
    ${transition(
      "border",
      theme.transitionTime.quick,
      theme.transitionTimingFunction.easeInOut
    )};
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  ${({ size }) => getSize(size)}
  ${({ disabled }) =>
    disabled &&
    `
      pointer-events: none;
  `}
  ${({ displaystyle, disabled }) => {
    switch (displaystyle) {
      case DisplayStyle.Primary:
        return css`
          height: 4rem;
          border-radius: ${borderRadius("medium")};
          background: ${backgroundColor("primary")};
          border: 0.1rem solid ${backgroundColor("primary")};
          ${disabled && disabledPrimaryStyles};
          &:hover {
            background: ${backgroundColor("primaryHover")};
          }
          g {
            fill: ${textColor("active")};
          }
          p {
            color: ${textColor("active")};
          }
        `
      case DisplayStyle.Secondary:
        return css`
          height: 4rem;
          border-radius: ${borderRadius("medium")};
          border: 0.1rem solid ${borderColor("primary")};
          g {
            fill: ${textColor("primary")};
          }
          ${disabled && disabledSecondaryStyles};
          &:hover {
            background: ${backgroundColor("secondaryHover")};
          }
        `
      case DisplayStyle.IconOnly1:
        return css`
          height: 4rem;
          width: 4rem;
          border: 0.1rem solid ${borderColor("hover")};
          border-radius: ${borderRadius("small")};
          &:hover {
            border-color: ${borderColor("primary")};
          }
        `
      case DisplayStyle.IconOnly2:
        return css`
          height: 3.2rem;
          width: 3.2rem;
          border-radius: ${borderRadius("small")};
          background: transparent;
          border: none;
          &:hover {
            background: ${backgroundColor("minor")};
          }

          svg {
            height: initial;
            width: initial;
          }
        `
      case DisplayStyle.IconOnly3:
        return css`
          height: 3.2rem;
          width: 3.2rem;
          border-radius: ${borderRadius("small")};
          background: transparent;
          border: none;
          &:hover {
            background: ${backgroundColor("minor")};
          }
          g {
            fill: ${textColor("action")};
          }
        `
      case DisplayStyle.Link1:
        return css`
          justify-content: flex-start;
          height: 3rem;
          padding: 0.8rem;
          border: none;
          border-radius: ${borderRadius("small")};
          font-weight: ${fontWeight("default")};
          width: 100%;
          &:hover {
            background-color: ${backgroundColor("minor")};
          }
        `
      case DisplayStyle.Link2:
        return css`
          justify-content: flex-start;
          height: 4rem;
          padding: 0.8rem;
          border: none;
          border-radius: ${borderRadius("medium")};
          font-weight: ${fontWeight("default")};
          width: 100%;
          &:hover {
            background-color: ${backgroundColor("minor")};
          }
        `
      case DisplayStyle.Link3:
        return css`
          justify-content: flex-start;
          height: 4rem;
          padding: 0.8rem;
          border: none;
          border-radius: ${borderRadius("small")};
          font-weight: ${fontWeight("default")};
          width: 100%;
          &:hover {
            background-color: ${backgroundColor("minor")};
          }
          g {
            fill: ${textColor("action")};
          }
          p {
            color: ${textColor("action")};
          }
        `
      case DisplayStyle.Link4:
        return css`
          justify-content: flex-start;
          height: 4rem;
          padding: 0.8rem;
          border: none;
          border-radius: ${borderRadius("small")};
          font-weight: ${fontWeight("default")};
          width: 100%;
          &.${activeClassName} {
            ${navLinkStyles}
          }
          &:hover {
            ${navLinkStyles}
          }
          svg {
            opacity: 0.6;
          }
          p {
            color: ${textColor("secondary")};
          }
        `
      case DisplayStyle.Tab:
        return css`
          justify-content: flex-start;
          height: 100%;
          padding: 0.8rem;
          border: none;
          position: relative;
          border-radius: ${borderRadius("small")}rem;
          font-weight: ${fontWeight("default")};
          width: 100%;

          &:after {
            content: "";
            position: absolute;
            display: block;
            bottom: 0;
            left: 0;
            width: 0;
            height: 0.2rem;
            background-color: ${backgroundColor("super")};
            transition: ${transition("width", undefined, "ease")};
          }

          :hover {
            color: ${textColor("tabHover")};
            svg {
              opacity: 0.9;
            }
          }

          p {
            color: ${textColor("secondary")};
          }

          &.${activeClassName} {
            color: ${textColor("primary")};
            &:after {
              width: 100%;
            }
            svg {
              opacity: 1;
            }
          }
          svg {
            opacity: 0.75;
          }
        `
      case DisplayStyle.Dropdown:
        return css`
          justify-content: flex-start;
          height: 3.2rem;
          padding: 0 1.1rem;
          border: none;
          border-radius: ${borderRadius("small")};
          font-weight: ${fontWeight("default")};
          width: 100%;
          min-width: 17.6rem;

          &.${activeClassName} {
            ${navLinkStyles}
          }
          &:hover {
            ${navLinkStyles}
          }
          svg {
            opacity: 0.75;
          }
          p {
            color: ${textColor("secondary")};
          }
        `
      case DisplayStyle.InputIcon:
        return css`
          justify-content: flex-start;
          height: 2.4rem;
          width: 2.4rem;
          border: none;
          padding: 0;
          ${disabled && {
            opacity: 0.4,
          }};
        `
      default:
        return
    }
  }}
`

export const StyledNavLink = styled(NavLink)<{
  displaystyle: DisplayStyle
  disabled: boolean
  size: Size
}>`
  ${buttonStyles}
`

export const StyledLink = styled(Link)<{
  displaystyle: DisplayStyle
  disabled: boolean
  size: Size
}>`
  ${buttonStyles}
`
export const StyledA = styled.a<{
  displaystyle: DisplayStyle
  disabled: boolean
  size: Size
}>`
  ${buttonStyles}
`
export const StyledButton = styled.button<{
  displaystyle: DisplayStyle
  disabled: boolean
  size: Size
}>`
  ${buttonStyles}
`
export const StyledIcon = styled(Icon)<{
  displaystyle: DisplayStyle
  withMargin: boolean
}>`
  ${({ displaystyle, withMargin }) => {
    if (withMargin) {
      if (displaystyle === DisplayStyle.Link2) {
        return css`
          margin: 0 1.2rem 0 0;
        `
      }
      return css`
        margin: 0 0.8rem 0 0;
      `
    }
    return css``
  }}

  ${({ displaystyle }) =>
    displaystyle === DisplayStyle.InputIcon &&
    css`
      width: 100%;
      height: 100%;
    `};
`

export const buttonComponentAnimationStyles = css`
  opacity: 0.6;
  transition: opacity ${transitionTime("quick")}
    ${transitionTimingFunction("smooth")};

  &:hover {
    opacity: 1;
  }
`
