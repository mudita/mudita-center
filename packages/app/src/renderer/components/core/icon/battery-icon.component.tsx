/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

import { FunctionComponent } from "Renderer/types/function-component.interface"
import * as React from "react"
import Icon, {
  Props as IconProps,
} from "Renderer/components/core/icon/icon.component"
import { Type } from "Renderer/components/core/icon/icon.config"

export interface BatteryIconProps {
  level: number
  charging?: boolean
}

const getInteractiveBatteryIcon = (
  batteryLevel: number,
  charging: boolean,
  size: number,
  rest: Partial<IconProps>
) => {
  switch (true) {
    case charging:
      return <Icon type={Type.ChargingBattery} {...rest} />
    case batteryLevel === 1:
      return <Icon type={Type.FullBattery} {...rest} />
    case batteryLevel > 0.8:
      return <Icon type={Type.VeryHighBattery} {...rest} />
    case batteryLevel > 0.6:
      return <Icon type={Type.HighBattery} {...rest} />
    case batteryLevel > 0.4:
      return <Icon type={Type.MediumBattery} {...rest} />
    case batteryLevel > 0.2:
      return <Icon type={Type.LowBattery} {...rest} />
    case batteryLevel > 0:
      return <Icon type={Type.VeryLowBattery} {...rest} />
    default:
      return <Icon type={Type.NoBattery} {...rest} />
  }
}

const BatteryIcon: FunctionComponent<BatteryIconProps & IconProps> = ({
  level,
  charging = false,
  height = 2,
  ...rest
}) => {
  return <>{getInteractiveBatteryIcon(level, charging, height, rest)}</>
}

export default BatteryIcon
