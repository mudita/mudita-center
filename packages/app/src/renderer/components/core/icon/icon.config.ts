/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { Image as ImageInterface } from "Renderer/interfaces/image.interface"
import Arrow from "Renderer/svg/arrow.svg"
import ArrowLongLeft from "Renderer/svg/arrow-long-left.svg"
import ArrowLongRight from "Renderer/svg/arrow-long-right.svg"
import AttachContact from "Renderer/svg/attach-contact.svg"
import BackupFolder from "Renderer/svg/backup-folder.svg"
import Battery from "Renderer/svg/battery.svg"
import BorderCheck from "Renderer/svg/border-check-icon.svg"
import ChargingBattery from "Renderer/svg/charging-battery.svg"
import Check from "Renderer/svg/check-icon.svg"
import CheckCircle from "Renderer/svg/check-circle.svg"
import CheckIndeterminate from "Renderer/svg/check-indeterminate.svg"
import Close from "Renderer/svg/close.svg"
import Delete from "Renderer/svg/delete.svg"
import FilesManager from "Renderer/svg/files-manager.svg"
import HighBattery from "Renderer/svg/high-battery.svg"
import HighRange from "Renderer/svg/high-range.svg"
import LowBattery from "Renderer/svg/low-battery.svg"
import LowRange from "Renderer/svg/low-range.svg"
import Magnifier from "Renderer/svg/magnifier.svg"
import Calendar from "Renderer/svg/menu-calendar.svg"
import MediumBattery from "Renderer/svg/medium-battery.svg"
import MediumRange from "Renderer/svg/medium-range.svg"
import MenuFilesManager from "Renderer/svg/menu-files-manager.svg"
import MenuHelp from "Renderer/svg/menu-help.svg"
import MenuMeditation from "Renderer/svg/menu-meditation.svg"
import Message from "Renderer/svg/menu-messages.svg"
import MenuMusic from "Renderer/svg/menu-music.svg"
import MenuNews from "Renderer/svg/menu-news.svg"
import MenuOverview from "Renderer/svg/menu-overview.svg"
import MenuPhone from "Renderer/svg/menu-phone.svg"
import MenuSettings from "Renderer/svg/menu-settings.svg"
import MenuTethering from "Renderer/svg/menu-tethering.svg"
import MenuTools from "Renderer/svg/menu-tools.svg"
import ExternalLink from "Renderer/svg/external-link.svg"
import MuditaLogo from "Renderer/svg/mudita.svg"
import MuditaLogoWithText from "Renderer/svg/mudita_logo.svg"
import MuditaLogoVertical from "Renderer/svg/mudita-logo-vertical.svg"
import NoBattery from "Renderer/svg/no-battery.svg"
import NoRange from "Renderer/svg/no-range.svg"
import Refresh from "Renderer/svg/refresh.svg"
import Send from "Renderer/svg/send.svg"
import Sim from "Renderer/svg/sim.svg"
import Template from "Renderer/svg/template.svg"
import Upload from "Renderer/svg/upload.svg"
import UploadDark from "Renderer/svg/upload-dark.svg"
import Download from "Renderer/svg/download.svg"
import DownloadWhite from "Renderer/svg/download-white.svg"
import Calls from "Renderer/svg/phone.svg"
import Connection from "Renderer/svg/connection.svg"
import Contact from "Renderer/svg/contact.svg"
import NewContact from "Renderer/svg/new-contact.svg"
import Notes from "Renderer/svg/notes.svg"
import Notifications from "Renderer/svg/notifications.svg"
import Dial from "Renderer/svg/dial.svg"
import Playlist from "Renderer/svg/playlist.svg"
import PlusSign from "Renderer/svg/plus-sign.svg"
import Reload from "Renderer/svg/circle-arrow.svg"
import VoiceRecorder from "Renderer/svg/voice-recorder.svg"
import VeryHighBattery from "Renderer/svg/very-high-battery.svg"
import Pure from "Renderer/svg/pure.svg"
import Fail from "Renderer/svg/fail.svg"
import FailRed from "Renderer/svg/fail-red.svg"
import VeryHighRange from "Renderer/svg/very-high-range.svg"
import VeryLowBattery from "Renderer/svg/very-low-battery.svg"
import Info from "Renderer/svg/info.svg"
import More from "Renderer/svg/more.svg"
import Blocked from "Renderer/svg/blocked.svg"
import Forward from "Renderer/svg/forward.svg"
import Edit from "Renderer/svg/edit.svg"
import Favourites from "Renderer/svg/favourites.svg"
import Ice from "Renderer/svg/ice.svg"
import ArrowDown from "Renderer/svg/arrow-down.svg"
import DeleteBig from "Renderer/svg/delete-big.svg"
import Attachment from "Renderer/svg/attachment.svg"
import Remove from "Renderer/svg/remove.svg"
import SendButton from "Renderer/svg/send-button.svg"
import MenuContacts from "Renderer/svg/menu-contacts.svg"
import Backup from "Renderer/svg/backup.svg"
import Tooltip from "Renderer/svg/tooltip.svg"
import TetheringStatus from "Renderer/svg/tethering-status.svg"
import Templates from "Renderer/svg/templates.svg"
import MissedCall from "Renderer/svg/missed-call.svg"
import ConferenceCall from "Renderer/svg/conference-call.svg"
import IncomingCall from "Renderer/svg/incoming-call.svg"
import OutgoingCall from "Renderer/svg/outgoing-call.svg"
import AppleLogo from "Renderer/svg/apple.svg"
import GoogleLogo from "Renderer/svg/google.svg"
import SynchronizeContacts from "Renderer/svg/synchronize-contacts.svg"
import MusicGrey from "Renderer/svg/music-grey.svg"
import VoiceRecorderGrey from "Renderer/svg/voice-recorder-gray.svg"
import FullBattery from "Renderer/svg/full-battery.svg"
import VeryLowRange from "Renderer/svg/very-low-range.svg"
import VeryHighRangeWithRoaming from "Renderer/svg/very-high-range-roaming.svg"
import HighRangeWithRoaming from "Renderer/svg/high-range-roaming.svg"
import MediumRangeWithRoaming from "Renderer/svg/medium-range-roaming.svg"
import LowRangeWithRoaming from "Renderer/svg/low-range-roaming.svg"
import VeryLowRangeWithRoaming from "Renderer/svg/very-low-range-roaming.svg"
import CalendarIcon from "Renderer/svg/calendar.svg"
import ThinCheck from "Renderer/svg/thin-check.svg"
import LongArrow from "Renderer/svg/long-arrow.svg"
import MarkAsRead from "Renderer/svg/mark-as-read.svg"
import RecoveryBackup from "Renderer/svg/recovery-backup.svg"
import RebootOs from "Renderer/svg/reboot-os.svg"
import FactoryReset from "Renderer/svg/factory-reset.svg"
import MuditaFlower from "Renderer/svg/mudita-flower.svg"
import MuditaDarkLogo from "Renderer/svg/mudita-dark-logo.svg"
import ContactGoogle from "Renderer/svg/contact-google.svg"
import Manage from "Renderer/svg/manage.svg"
import MuditaLogoBg from "Renderer/svg/muditalogo-bg.svg"
import Outlook from "Renderer/svg/outlook.svg"
import ContactFilled from "Renderer/svg/contact-filled.svg"
import ThinFail from "Renderer/svg/thin-fail.svg"
import Support from "Renderer/svg/support.svg"
import Visible from "Renderer/svg/password-show.svg"
import Hidden from "Renderer/svg/password-hide.svg"
import Cloud from "Renderer/svg/cloud.svg"
import { FunctionComponent } from "Renderer/types/function-component.interface"
import { IconType } from "Renderer/components/core/icon/icon-type"

const typeToIcon: Partial<Record<IconType, typeof Arrow>> = {
  [IconType.Arrow]: Arrow,
  [IconType.ArrowLongLeft]: ArrowLongLeft,
  [IconType.ArrowLongRight]: ArrowLongRight,
  [IconType.AttachContact]: AttachContact,
  [IconType.BackupFolder]: BackupFolder,
  [IconType.Battery]: Battery,
  [IconType.BorderCheckIcon]: BorderCheck,
  [IconType.Calendar]: Calendar,
  [IconType.CalendarIcon]: CalendarIcon,
  [IconType.ChargingBattery]: ChargingBattery,
  [IconType.Check]: Check,
  [IconType.CheckCircle]: CheckCircle,
  [IconType.CheckIndeterminate]: CheckIndeterminate,
  [IconType.Close]: Close,
  [IconType.Cloud]: Cloud,
  [IconType.Delete]: Delete,
  [IconType.FilesManager]: FilesManager,
  [IconType.HighBattery]: HighBattery,
  [IconType.HighRange]: HighRange,
  [IconType.LowBattery]: LowBattery,
  [IconType.LowRange]: LowRange,
  [IconType.Magnifier]: Magnifier,
  [IconType.MediumBattery]: MediumBattery,
  [IconType.MenuContacts]: MenuContacts,
  [IconType.MediumRange]: MediumRange,
  [IconType.MenuFilesManager]: MenuFilesManager,
  [IconType.MenuHelp]: MenuHelp,
  [IconType.MenuMeditation]: MenuMeditation,
  [IconType.MenuMusic]: MenuMusic,
  [IconType.MenuNews]: MenuNews,
  [IconType.MenuOverview]: MenuOverview,
  [IconType.MenuPhone]: MenuPhone,
  [IconType.MenuSettings]: MenuSettings,
  [IconType.MenuTethering]: MenuTethering,
  [IconType.MenuTools]: MenuTools,
  [IconType.Message]: Message,
  [IconType.ExternalLink]: ExternalLink,
  [IconType.MuditaLogo]: MuditaLogo,
  [IconType.MuditaLogoWithText]: MuditaLogoWithText,
  [IconType.MuditaLogoVertical]: MuditaLogoVertical,
  [IconType.NoBattery]: NoBattery,
  [IconType.NoRange]: NoRange,
  [IconType.Refresh]: Refresh,
  [IconType.Send]: Send,
  [IconType.Sim]: Sim,
  [IconType.Support]: Support,
  [IconType.Template]: Template,
  [IconType.Tooltip]: Tooltip,
  [IconType.TetheringStatus]: TetheringStatus,
  [IconType.Templates]: Templates,
  [IconType.Calls]: Calls,
  [IconType.Connection]: Connection,
  [IconType.Contact]: Contact,
  [IconType.NewContact]: NewContact,
  [IconType.Dial]: Dial,
  [IconType.Notes]: Notes,
  [IconType.Notifications]: Notifications,
  [IconType.Playlist]: Playlist,
  [IconType.PlusSign]: PlusSign,
  [IconType.Reload]: Reload,
  [IconType.Upload]: Upload,
  [IconType.UploadDark]: UploadDark,
  [IconType.VeryHighBattery]: VeryHighBattery,
  [IconType.VeryHighRange]: VeryHighRange,
  [IconType.VeryLowBattery]: VeryLowBattery,
  [IconType.VoiceRecorder]: VoiceRecorder,
  [IconType.Info]: Info,
  [IconType.Download]: Download,
  [IconType.DownloadWhite]: DownloadWhite,
  [IconType.Pure]: Pure,
  [IconType.Fail]: Fail,
  [IconType.FailRed]: FailRed,
  [IconType.More]: More,
  [IconType.Blocked]: Blocked,
  [IconType.Forward]: Forward,
  [IconType.Edit]: Edit,
  [IconType.Favourites]: Favourites,
  [IconType.Ice]: Ice,
  [IconType.ArrowDown]: ArrowDown,
  [IconType.DeleteBig]: DeleteBig,
  [IconType.Attachment]: Attachment,
  [IconType.Remove]: Remove,
  [IconType.SendButton]: SendButton,
  [IconType.Backup]: Backup,
  [IconType.Apple]: AppleLogo,
  [IconType.Google]: GoogleLogo,
  [IconType.SynchronizeContacts]: SynchronizeContacts,
  [IconType.MissedCall]: MissedCall,
  [IconType.ConferenceCall]: ConferenceCall,
  [IconType.IncomingCall]: IncomingCall,
  [IconType.OutgoingCall]: OutgoingCall,
  [IconType.MusicGrey]: MusicGrey,
  [IconType.VoiceRecorderGrey]: VoiceRecorderGrey,
  [IconType.FullBattery]: FullBattery,
  [IconType.VeryLowRange]: VeryLowRange,
  [IconType.VeryHighRangeWithRoaming]: VeryHighRangeWithRoaming,
  [IconType.HighRangeWithRoaming]: HighRangeWithRoaming,
  [IconType.MediumRangeWithRoaming]: MediumRangeWithRoaming,
  [IconType.LowRangeWithRoaming]: LowRangeWithRoaming,
  [IconType.VeryLowRangeWithRoaming]: VeryLowRangeWithRoaming,
  [IconType.MarkAsRead]: MarkAsRead,
  [IconType.ThinCheck]: ThinCheck,
  [IconType.MuditaFlower]: MuditaFlower,
  [IconType.LongArrow]: LongArrow,
  [IconType.RecoveryBackup]: RecoveryBackup,
  [IconType.RebootOs]: RebootOs,
  [IconType.FactoryReset]: FactoryReset,
  [IconType.MuditaDarkLogo]: MuditaDarkLogo,
  [IconType.ContactGoogle]: ContactGoogle,
  [IconType.MuditaLogoBg]: MuditaLogoBg,
  [IconType.Manage]: Manage,
  [IconType.Outlook]: Outlook,
  [IconType.ContactFilled]: ContactFilled,
  [IconType.Visible]: Visible,
  [IconType.Hidden]: Hidden,
  [IconType.ThinFail]: ThinFail,
}

export const getIconType = (
  icon: IconType = IconType.Message
): FunctionComponent<ImageInterface> => typeToIcon[icon] || Message

export function getEnumName(type: IconType): string
export function getEnumName(type?: IconType): null
export function getEnumName(type?: IconType): string | null {
  return type !== undefined ? IconType[type] : null
}
