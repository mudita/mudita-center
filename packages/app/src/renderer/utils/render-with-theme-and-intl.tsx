/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "Renderer/locales/default/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"
import ReactModal from "react-modal"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
