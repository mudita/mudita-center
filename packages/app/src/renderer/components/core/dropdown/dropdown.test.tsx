/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { fireEvent } from "@testing-library/dom"
import "@testing-library/jest-dom"
import React from "react"
import ButtonComponent from "Renderer/components/core/button/button.component"
import { DisplayStyle } from "Renderer/components/core/button/button.config"
import Dropdown from "Renderer/components/core/dropdown/dropdown.component"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import Button from "../button/button.component"
import { Type } from "Renderer/components/core/icon/icon.config"

test("matches snapshot", () => {
  const { container } = renderWithThemeAndIntl(
    <Dropdown toggler={<ButtonComponent />} />
  )
  expect(container).toMatchSnapshot()
})

test("renders toggler passed to component", () => {
  const buttonText = "Example"
  const { getByText } = renderWithThemeAndIntl(
    <Dropdown toggler={<ButtonComponent label={buttonText} />} />
  )
  expect(getByText(buttonText)).toBeInTheDocument()
})

test("renders dropdown", () => {
  const buttonText = "Example"
  const { getByTestId, getByText, container } = renderWithThemeAndIntl(
    <Dropdown toggler={<ButtonComponent label={buttonText} />}>
      <Button
        displayStyle={DisplayStyle.Link1}
        label="I open Google in new tab"
        href="http://www.google.pl"
        target="_blank"
        Icon={Type.Upload}
      />
    </Dropdown>
  )

  fireEvent.click(getByText(buttonText))

  expect(getByTestId("dropdown")).toBeVisible()
  expect(container).toMatchSnapshot()
})

test("renders children", () => {
  const buttonText = "Example"
  const childText = "childText"
  const { getByText } = renderWithThemeAndIntl(
    <Dropdown toggler={<ButtonComponent label={buttonText} />}>
      <Button
        displayStyle={DisplayStyle.Link1}
        label={childText}
        href="http://www.google.pl"
        target="_blank"
        Icon={Type.Upload}
      />
    </Dropdown>
  )

  fireEvent.click(getByText(buttonText))

  expect(getByText(childText)).toBeVisible()
})
