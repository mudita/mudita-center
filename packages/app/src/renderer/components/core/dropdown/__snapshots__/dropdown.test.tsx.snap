// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  color: #ffffff;
  border-radius: 0.5rem;
  background: #3b3f42;
  border: 0.1rem solid #3b3f42;
}

.c1:hover {
  background: #232426;
}

.c1 g {
  fill: #ffffff;
}

.c0 {
  position: relative;
  z-index: 0;
}

.c2 {
  position: absolute;
  list-style-type: none;
  padding: 1.4rem 0;
  background-color: #ffffff;
  border-radius: 0.2rem 0.2rem;
  box-shadow: 0 1rem 5.5rem -0.5rem rgba(188,188,188,0.5);
  min-width: 17rem;
  pointer-events: none;
  opacity: 0;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
  right: 0;
  top: 100%;
  margin-top: 1rem;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      size="2"
      type="button"
    />
    <ul
      class="c2"
      data-testid="dropdown"
    />
  </div>
</div>
`;

exports[`renders dropdown 1`] = `
.c2 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 2.8rem;
  width: 2.8rem;
  position: relative;
  font-style: normal;
}

.c6 svg {
  width: 100%;
  height: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  height: 3rem;
  padding: 0.8rem;
  border: none;
  border-radius: 0.2rem;
  font-weight: 400;
  width: 100%;
}

.c5:hover {
  background-color: #f4f5f6;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  color: #ffffff;
  border-radius: 0.5rem;
  background: #3b3f42;
  border: 0.1rem solid #3b3f42;
}

.c1:hover {
  background: #232426;
}

.c1 g {
  fill: #ffffff;
}

.c7 {
  margin: 0 0.8rem 0 0;
}

.c3 {
  white-space: nowrap;
}

.c0 {
  position: relative;
  z-index: 3;
}

.c4 {
  position: absolute;
  list-style-type: none;
  padding: 1.4rem 0;
  background-color: #ffffff;
  border-radius: 0.2rem 0.2rem;
  box-shadow: 0 1rem 5.5rem -0.5rem rgba(188,188,188,0.5);
  min-width: 17rem;
  pointer-events: auto;
  opacity: 1;
  -webkit-transition: opacity 500ms ease;
  transition: opacity 500ms ease;
  right: 0;
  top: 100%;
  margin-top: 1rem;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      size="2"
      type="button"
    >
      <p
        class="c2 c3"
      >
        Example
      </p>
    </button>
    <ul
      class="c4"
      data-testid="dropdown"
    >
      <a
        class="c5"
        href="http://www.google.pl"
        size="2"
        target="_blank"
      >
        <i
          class="c6 c7"
          data-testid="icon-Upload"
          height="2.8"
          width="2.8"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </i>
        <p
          class="c2 c3"
        >
          I open Google in new tab
        </p>
      </a>
    </ul>
  </div>
</div>
`;
